### ISSUE TITLE
kubefed join breaks with aws cluster

### ISSUE THREAD

**Issue by irfanurrehman** [kubernetes/federation#23]
<a href="https://github.com/henriquetruta"><img src="https://avatars2.githubusercontent.com/u/2501482?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [henriquetruta](https://github.com/henriquetruta)**
_Monday Jul 17, 2017 at 18:23 GMT_
_Originally opened as https://github.com/kubernetes/kubernetes/issues/49041_

----

Joining a cluster from AWS fails in kubefed 1.7. The AWS cluster is also k8s 1.7
The federation is running in a GKE cluster 1.7.0. 

```
$ kubefed join aws --host-cluster-context=gke --context=fed
Error from server (Forbidden): clusterroles.rbac.authorization.k8s.io "federation-controller-manager:fed-aws-gke" is forbidden: attempt to grant extra privileges: [PolicyRule{Resources:["*"], APIGroups:["*"], Verbs:["*"]} PolicyRule{NonResourceURLs:["/healthz"], Verbs:["get"]}] user=&{admin admin [system:authenticated] map[]} ownerrules=[] ruleResolutionErrors=[]
```

This same command works in kubefed 1.6.6
I'm able to access the AWS cluster using its context.

/kind bug

**How to reproduce it (as minimally and precisely as possible)**:
1 - Create a GKE cluster
2 - Create an AWS cluster (I used conjure-up)
3 - kubefed (1.7) init on GKE cluster
4 - Join the AWS cluster
5 - It should fail

If kubefed 1.6 is used, it works.
Joining a GKE or On-prem cluster also works as expected

**Comment by irfanurrehman** [comment]
<a href="https://github.com/henriquetruta"><img src="https://avatars2.githubusercontent.com/u/2501482?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [henriquetruta](https://github.com/henriquetruta)**
_Monday Jul 17, 2017 at 18:34 GMT_

----

/sig federation

**Comment by irfanurrehman** [comment]
<a href="https://github.com/quinton-hoole"><img src="https://avatars0.githubusercontent.com/u/10390785?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [quinton-hoole](https://github.com/quinton-hoole)**
_Saturday Sep 02, 2017 at 00:35 GMT_

----

cc @madhusudancs

**Comment by irfanurrehman** [comment]
<a href="https://github.com/GabrielSVinha"><img src="https://avatars2.githubusercontent.com/u/17187082?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [GabrielSVinha](https://github.com/GabrielSVinha)**
_Wednesday Sep 20, 2017 at 17:45 GMT_

----

I believe this is solved in 1.7+ where RBAC is default for all providers.

**Comment by irfanurrehman** [comment]
cc @henriquetruta
